src/AppSystem/Launcher.vala

```diff
@@ -30,7 +36,10 @@ public class Dock.Launcher : BaseItem {
     private Gtk.Revealer badge_revealer;
     private Adw.TimedAnimation bounce_up;
     private Adw.TimedAnimation bounce_down;
-    private Gtk.PopoverMenu popover;
+    private Gtk.PopoverMenu popover_menu;
+    private Gtk.Popover popover_tooltip;
+
+    private Gtk.EventControllerMotion motion_controller;
```

Comment:
I don't think we need to hold a reference to the event controllers in Gtk4. You can just make it a local variable and  `add_controller (motion_controller)` should handle the rest.